src/praisonai-agents/praisonaiagents/agent/agent.py

```
@@ -967,7 +1122,13 @@ def chat(self, prompt, temperature=0.2, tools=None, output_json=None, output_pyd
                     total_time = time.time() - start_time
                     logging.debug(f"Agent.chat completed in {total_time:.2f} seconds")
 
-                return response_text
+                # Apply guardrail validation for custom LLM response
+                try:
+                    validated_response = self._apply_guardrail_with_retry(response_text, prompt, temperature, tools)
```

Pylint (3.3.7)

[convention] Line too long (116/100)

(C0301)